src/base/controller.hpp

```diff
@@ -265,6 +278,18 @@ void Controller::set_parallelization(Qobj& qobj) {
   parallel_shots_ = std::max<int>({1, parallel_shots_});
 
   parallel_state_update_ = std::max<int>({1, max_parallel_threads_/(parallel_experiments_*parallel_shots_)});
+
+  if(state_available_memory_mb_ > 0)
+  {
+    if (parallel_experiments_ > 1)
+    {
+      state_available_memory_mb_ /= parallel_experiments_;
+    }
+    else if (parallel_shots_ > 1)
```

Comment:
What if `parallel_experiments_` are greater than 1 and `parallel_shots_` are greater than 1 as well?
We will never enter in the second conditional: `else if( parallel_shots_ > 1)` so I guess that the intention is to prioritize parallel experiments over parallel shots, isn't it?.
I could think that in this situation, we may want to divide twice: one by parallel shots, and the resulting available memory needs to be divided again by parallel experiments.

Reply:
This was based on line 229 earlier in the file, as part of `Controller::set_config`
```C++
if (max_parallel_eperiments_ > 1)
{
  max_parallel_shots_ = 1;
}
```
So at any time only one of parallel experients/shots will be greater than 1.